contracts/zero-ex/contracts/src/transformers/bridges/mixins/MixinPlatypus.sol

```diff
@@ -0,0 +1,93 @@
+// SPDX-License-Identifier: Apache-2.0
+/*
+
+  Copyright 2020 ZeroEx Intl.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.6.5;
+pragma experimental ABIEncoderV2;
+
+import "@0x/contracts-utils/contracts/src/v06/errors/LibRichErrorsV06.sol";
+import "@0x/contracts-erc20/contracts/src/v06/LibERC20TokenV06.sol";
+import "@0x/contracts-erc20/contracts/src/v06/IERC20TokenV06.sol";
+import "@0x/contracts-utils/contracts/src/v06/LibSafeMathV06.sol";
+
+contract MixinPlatypus {
+
+    using LibERC20TokenV06 for IERC20TokenV06;
+    using LibSafeMathV06 for uint256;
+    using LibRichErrorsV06 for bytes;
+
+    struct PlatypusBridgeData {
+        address platypusAddress;
+        address fromToken;
+        address toToken;
+        uint256 fromAmount;
+        uint256 minimumToAmount;
+        bytes4 exchangeFunctionSelector;
+        bytes4 quoteFunctionSelector;
+    }
+
+
+    function _swapQuote(
+        PlatypusBridgeData memory data
+    ) internal view returns (uint256 quotedAmountOut){
+        (bool success, bytes memory resultData) =
+            data.platypusAddress.staticcall(abi.encodeWithSelector(
+                data.quoteFunctionSelector,
+                data.fromToken,
+                data.toToken,
+                data.fromAmount
+        ));
+        if (!success) {
+            resultData.rrevert();
+        }
+        (uint256 quotedAmountOut, uint256 haircut) = abi.decode(resultData, (uint256, uint256));
+        return quotedAmountOut;
+    }
+
+    function _tradePlatypus(
+        IERC20TokenV06 sellToken,
+        IERC20TokenV06 buyToken,
+        uint256 sellAmount,
+        bytes memory bridgeData
+    )
+        internal
+        returns (uint256 boughtAmount)
+    {
+        // Decode the bridge data to get the Platypus metadata.
+        PlatypusBridgeData memory data = abi.decode(bridgeData, (PlatypusBridgeData));
+        sellToken.approveIfBelow(data.platypusAddress, sellAmount);
+
+        uint256 beforeBalance = buyToken.balanceOf(address(this));
+        uint256 minAmountOut = _swapQuote(data);
+        (bool success, bytes memory resultData) =
```

Comment:
I believe the swap function returns the output amount

Reply:
forgot it auto requests you to review this when i make a pr, im moving the new mixin into asset swapper and the platypus as a curve source was just testing from a bit ago to see if i could squeeze it into our already existing interfaces.

Reply:
and yeah, i can pass in a min amount of 0 here that will be removed